### PR TITLE
[SQL] Allow field access in nullable structs

### PR DESCRIPTION
**Pull request by mihaibudiu**

### CODE REVIEW

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/rust/ToRustInnerVisitor.java [hunk 1]
```diff
@@ -1206,17 +1206,33 @@ public VisitDecision preorder(DBSPCustomOrdField expression) {
...
         DBSPType baseType = expression.expression.getType();
         if (baseType.mayBeNull) {
-            // TODO: this should be done differently
-            if (!baseType.hasCopy() &&
-                    !expression.expression.is(DBSPCloneExpression.class))
-                this.builder.append(".clone()");
-            this.builder.append(".unwrap()");
+            // Accessing a nullable struct is allowed
+            if (!expression.getType().mayBeNull) {
+                // It the result is not null, we need to unwrap()
```
Comment (lalithsuresh) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/rust/ToRustInnerVisitor.java:1213]: If

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/StructTests.java [hunk 2]
```diff
@@ -253,4 +257,26 @@ lastname        VARCHAR(30),
         ccs.addPair(new Change(input), new Change(output));
         this.addRustTestCase(ccs);
     }
+
+    @Test
+    public void nullableStruct() {
```
Comment (lalithsuresh) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/StructTests.java:262]: Worth also testing one more level of user-defined type nesting.

**Comment by mihaibudiu** [comment]
This needs a review to be merged

**Review by ryzhyk — APPROVED** [review]
Fixes the panic for me

**Pull request merged**